torch/_dynamo/guards.py

```diff
@@ -430,22 +430,36 @@ def CONSTANT_MATCH(self, guard: Guard):
             self.EQUALS_MATCH(guard)
 
     def NN_MODULE(self, guard: Guard):
+        # TODO(janimesh) - This id_match can be removed because nn_module_guard
+        # checks this in C. However, we need this redundant check to allow
+        # flexible cache size policy when we guard on self of nn module
+        # instances. See Note in torch/_dynamo/cache_size.py
         self.ID_MATCH(guard)
```

Comment:
@colesbury This is a change from your PR. I need this because we want to limit the number of cache entries per ID_MATCH'd object. This ID_MATCH is now hidden inside the C `nn_module_guard` where we do `mod == saved_mod` check.